pkg/api/v1/reporter/reporter.go

```diff
@@ -46,7 +46,7 @@ func (e ResourceErrors) Validate() error {
 }
 
 type Reporter interface {
-	WriteReports(ctx context.Context, errs ResourceErrors, subresourceStatuses map[string]*core.Status) error
+	WriteReports(ctx context.Context, errs ResourceErrors, subresourceStatuses map[string]*core.Status, messages []string) error
```

Comment:
in our repos where the users are other repos (i.e. this is a library for another repo) we mark changes to public APIs such as this as breaking changes, since consumers (other code) will have to update so it compiles. thus the changelog here should be `BREAKING_CHANGE` and the changelog version should be updated accordingly

note that we don't do the same for Gloo / Solo-Projects (in those repos this would be non-breaking) since this would require a major version bump, something we only do once per quarter. Also, our users there are mostly people interacting with our CRs/yaml (which we should _never_ break), so this change is non-breaking to them

Reply:
Actually this is something i did incorrectly that I just rolled back but good to know for when it does come up!